src/test/java/io/wgj/ctci/StackTest.java

```diff
@@ -0,0 +1,50 @@
+package io.wgj.ctci;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class StackTest {
+    @Test
+    public void testStackOneInt() {
+        Stack s = new Stack();
+        s.push(1);
+        int r = s.pop();
+        assertEquals(1, r);
+    }
+
+    @Test
+    public void testStackMultipleInts() {
+        ArrayList<Integer> expected = new ArrayList<>();
+        for (int i = 2; i < 10; i = i + 2) {
```

Comment:
Pedantic: When I look at a test, I expect everything to be straightoward and reasonable.  When I look at this, my first thought is "why does this iterate from 2, 4, ...., 18 instead of 0, 1, ...., 9"?  IMHO things like this make it harder to understand what a test is supposed to be doing because you have to think twice about the intention.

That is, assuming there isn't *actually* a reason.  If thats the case it should have a comment.

Reply:
Broke into two tests, one for natural sequences and one for random number, which I think better shows my intent, such that an obvious sequence of 1..10 wouldn't hide a bug.